design/use_cases/keep_out_zones_use_case.md

```diff
@@ -0,0 +1,15 @@
+# Keep Out Zones
+As a Robot user I want to be able to designate keep-out zones or areas on a map so that my robot will go around those areas instead of through them 
+
+## More details
+- Why is this needed?
+   - This is needed for indoor/outdoor robot navigation in areas where there may be safety issues or hazards
+   - Example: a courier robot in a logistics warehouse may be required to avoid areas where the forklift is unloading pallets from trucks
+
+- What is the expected user interaction?
+   - The user should be able to specify a keep out zones on a map through a GUI 
+   - The keep out zones should also be able to be specified programmatically through an API or config file
+
+- Are there any non-functional requirements? (build system, tools, performance, etc)
```

Comment:
If this is just a map, then Image editors can be used to make keepout zones. (we did this on the fetch research robot -- https://github.com/fetchrobotics/fetch_ros/tree/indigo-devel/fetch_maps/maps). If it is some config file or API, we probably need a graphical tool for this to be useful to most people.

Reply:
The GUI could simply be a Rviz plugin.